numba/core/withcontexts.py

```diff
@@ -220,12 +222,22 @@ def _legalize_args(self, args, kwargs, loc, func_globals, func_closures):
                         f"Global {v.name!r} is not defined"
                     )
                 typeanns[k] = v
+            elif isinstance(v, ir.Expr) and v.op == "getattr":
+                try:
+                    base_obj = func_ir.infer_constant(v.value)
+                    typ = getattr(base_obj, v.attr)
+                except (errors.ConstantInferenceError, AttributeError):
+                    raise errors.CompilerError(
+                        f"Getattr cannot be resolved at compile-time",
```

Comment:
Should this include information from `base_obj` and `v.value`? From e.g. this:
```python
from numba import njit, objmode, types
import numpy as np

@njit
def moderror():
    with objmode(v1=types.NOT_A_TYPE, v2=types.intp, v3=types.float32):
        v1 = 12.3
        v2 = 12.3
        v3 = 12.3
    return v1, v2, v3

moderror()
```
gives:
```
numba.core.errors.CompilerError: Failed in nopython mode pipeline (step: Handle with contexts)
Getattr cannot be resolved at compile-time

File "test_6680.py", line 6:
def moderror():
    with objmode(v1=types.NOT_A_TYPE, v2=types.intp, v3=types.float32):
    ^
```
whilst entirely obvious in this contrived case, in general would it be clear which of the `v{1,2,3}` is the issue?

Reply:
I've made the msg better. The above will now show:

```
CompilerError: Failed in nopython mode pipeline (step: Handle with contexts)
Error handling objmode argument 'v1'. Getattr cannot be resolved at compile-time.

File "<ipython-input-1-a794355d32dc>", line 6:
def moderror():
    with objmode(v1=types.NOT_A_TYPE, v2=types.intp, v3=types.float32):
    ^
```

Reply:
Thanks for changing that, the latest patch makes this case much clearer.